src/components/structures/ScrollPanel.js

```diff
@@ -268,6 +329,42 @@ module.exports = React.createClass({
         }
     },
 
+    // check if unfilling is possible and send an unfill request if necessary
+    _checkUnfillState: function(backwards) {
+        let excessHeight = this.getExcessHeight(backwards);
+        if (excessHeight > 0) {
+            var itemlist = this.refs.itemlist;
+            var tiles = itemlist.children;
+
+            // The scroll token of the first/last tile to be unpaginated
+            let markerScrollToken = null;
+
+            // Subtract clientHeights to simulate the events being unpaginated whilst counting
+            // the events to be unpaginated.
+            if (backwards) {
+                // Iterate forwards from start of tiles, subtracting event tile height
+                let i = 0;
+                while (i < tiles.length && excessHeight > tiles[i].clientHeight) {
+                    excessHeight -= tiles[i].clientHeight;
+                    markerScrollToken = tiles[i].dataset.scrollToken;
```

Comment:
not all tiles have a scrolltoken (dateseparators in particular do not iirc). you should probably preserve the last known `markerScrollToken` in that situation.